services/motion/explore/explore.go

```diff
@@ -394,28 +385,26 @@ func (ms *explore) checkForObstacles(
 
 // executePlan will carry out the desired motionplan plan.
 func (ms *explore) executePlan(ctx context.Context, kb kinematicbase.KinematicBase, plan motionplan.Plan) {
-	// Iterate through motionplan plan
-	for _, p := range plan {
+	steps, err := plan.Trajectory().GetFrameInputs(kb.Name().Name)
+	if err != nil {
+		return
+	}
+	for _, inputs := range steps {
 		select {
 		case <-ctx.Done():
 			return
 		default:
-			if inputEnabledKb, ok := kb.(inputEnabledActuator); ok {
-				if err := inputEnabledKb.GoToInputs(ctx, p[kb.Name().Name]); err != nil {
-					// If there is an error on GoToInputs, stop the component if possible before returning the error
-					if stopErr := kb.Stop(ctx, nil); stopErr != nil {
-						ms.executionResponseChan <- moveResponse{err: err}
-						return
-					}
-					// If the error was simply a cancellation of context return without erroring out
-					if errors.Is(err, context.Canceled) {
-						return
-					}
+			if err := kb.GoToInputs(ctx, inputs); err != nil {
```

Comment:
Is this fixing a bug?

Reply:
There was no need to change this after all.  Will revert